src/devices/UFireIse/UFireIse.cs

```diff
@@ -0,0 +1,391 @@
+
+            DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+
+            _measurement = ReadRegister((byte)Register.ISE_MV_REGISTER);
+            _measurement = Convert.ToSingle(Math.Round(_measurement, 2));
+
+            return new ElectricPotential(_measurement, UnitsNet.Units.ElectricPotentialUnit.Millivolt);
+        }
+
+        /// <summary>
+        /// Measure temperature
+        /// </summary>
+        /// <returns>Temperature. Temperature range: -2 to 35 C. Temperature Precision: 0.05 C</returns>
+        public Temperature MeasureTemperature()
+        {
+            SendCommand((byte)Command.ISE_MEASURE_TEMP);
+
+            DelayHelper.DelayMilliseconds(IseTemperatureMeasureTime, allowThreadYield: true);
+            _temperature = new Temperature(ReadRegister((byte)Register.ISE_TEMP_REGISTER), UnitsNet.Units.TemperatureUnit.DegreeCelsius);
+
+            return _temperature;
+        }
+
+        /// <summary>
+        /// Set temperature compensation
+        /// </summary>
+        /// <param name="temp">Compensation temperature</param>
+        public void SetTemperatureCompensation(Temperature temp)
+        {
+            WriteRegister((byte)Register.ISE_TEMP_REGISTER, Convert.ToSingle(temp.DegreesCelsius));
+            _temperature = temp;
+        }
+
+        /// <summary>
+        /// Calibrates the probe using a single point using a mV value.
+        /// </summary>
+        /// <param name="solution">Value. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateSingle(ElectricPotential solution)
+        {
+            if (solution.Millivolts >= -1024 && solution.Millivolts <= 1024)
+            {
+                WriteRegister((byte)Register.ISE_SOLUTION_REGISTER, Convert.ToSingle(solution.Millivolts));
+                SendCommand((byte)Command.ISE_CALIBRATE_SINGLE);
+
+                DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+            }
+        }
+
+        /// <summary>
+        /// Calibrates the dual-point values for the low reading, in mV, and saves them in the devices's EEPROM.
+        /// </summary>
+        /// <param name="solution">Value. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateProbeLow(ElectricPotential solution)
+        {
+            if (solution.Millivolts >= -1024 && solution.Millivolts <= 1024)
+            {
+                WriteRegister((byte)Register.ISE_SOLUTION_REGISTER, Convert.ToSingle(solution.Millivolts));
+                SendCommand((byte)Command.ISE_CALIBRATE_LOW);
+
+                DelayHelper.DelayMilliseconds(IseMvMeasureTime, allowThreadYield: true);
+            }
+        }
+
+        /// <summary>
+        /// Calibrates the dual-point values for the high reading, in mV, and saves them in the devices's EEPROM.
+        /// </summary>
+        /// <param name="solution">Value. Range: -1024 mV to 1024 mV</param>
+        public void CalibrateProbeHigh(ElectricPotential solution)
```

Comment:
can we merge three methods into one with longer explanation? I assume you need to pass all 3 points in order to make this usable

Reply:
(internally you can have 3 methods but public API should be more obvious how to use)